src/List.js

```diff
@@ -1,11 +1,41 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import {
+  collection,
+  getDocs,
+  query,
+  onSnapshot,
+  doc,
+  setDoc,
+} from '@firebase/firestore';
+import { db } from './lib/firebase.js';
 
 export function List() {
+  const [items, setItems] = useState([]);
+
+  useEffect(() => {
+    const fetchItems = async () => {
+      const response = collection(db, 'shopping-list');
+      const itemList = query(response);
+
+      const unsubscribe = onSnapshot(itemList, (querySnapshot) => {
+        const items = querySnapshot.docs.reduce((acc, doc) => {
+          const { name } = doc.data();
+          const id = doc.id;
+          return [...acc, { id, name }];
```

Comment:
I have a question, I think that in the UseEffect you are not searching by token, which I believe means that you'll get all the items in the database. Can you test creating an item for a different list with another token, if you see all the items that are in the database?

Reply:
Yes, maybe it isn't you scope in this issue . Thanks for doing that test for me!!